central/group/datastore/datastore_impl.go

```diff
@@ -174,6 +174,10 @@ func (ds *dataStoreImpl) Mutate(ctx context.Context, remove, update, add []*stor
 			// Use the id of the retrieved group to delete
 			propsID = id
 		}
+		err := ds.validateMutableGroupID(ctx, propsID, force)
```

Comment:
I'd say it's time to create a function similar to `validateAndPrepGroupForUpdateNoLock`, which will handle validation of group, retrieval of ID etc. for the delete case.